tools/src/test/java/org/apache/kafka/tools/reassign/ReassignPartitionsUnitTest.java

```diff
@@ -0,0 +1,787 @@
+import static kafka.admin.ReassignPartitionsCommand.getReplicaAssignmentForTopics;
+import static kafka.admin.ReassignPartitionsCommand.modifyInterBrokerThrottle;
+import static kafka.admin.ReassignPartitionsCommand.modifyLogDirThrottle;
+import static kafka.admin.ReassignPartitionsCommand.modifyTopicThrottles;
+import static kafka.admin.ReassignPartitionsCommand.parseExecuteAssignmentArgs;
+import static kafka.admin.ReassignPartitionsCommand.parseGenerateAssignmentArgs;
+import static kafka.admin.ReassignPartitionsCommand.partitionReassignmentStatesToString;
+import static kafka.admin.ReassignPartitionsCommand.replicaMoveStatesToString;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+
+@Timeout(60)
+public class ReassignPartitionsUnitTest {
+    @BeforeEach
+    public void setUp() {
+        Exit.setExitProcedure((statusCode, message) -> {
+            throw new IllegalArgumentException(message);
+        });
+    }
+
+    @AfterEach
+    public void tearDown() {
+        Exit.resetExitProcedure();
+    }
+
+    @Test
+    public void testCompareTopicPartitions() {
+        assertTrue(compareTopicPartitions(new TopicPartition("abc", 0),
+            new TopicPartition("abc", 1)));
+        assertFalse(compareTopicPartitions(new TopicPartition("def", 0),
+            new TopicPartition("abc", 1)));
+    }
+
+    @Test
+    public void testCompareTopicPartitionReplicas() {
+        assertTrue(compareTopicPartitionReplicas(new TopicPartitionReplica("def", 0, 0),
+            new TopicPartitionReplica("abc", 0, 1)));
+        assertFalse(compareTopicPartitionReplicas(new TopicPartitionReplica("def", 0, 0),
+            new TopicPartitionReplica("cde", 0, 0)));
+    }
+
+    @Test
+    public void testPartitionReassignStatesToString() {
+        Map<TopicPartition, ReassignPartitionsCommand.PartitionReassignmentState> states = new HashMap<>();
+
+        states.put(new TopicPartition("foo", 0),
+            new ReassignPartitionsCommand.PartitionReassignmentState(seq(1, 2, 3), seq(1, 2, 3), true));
+        states.put(new TopicPartition("foo", 1),
+            new ReassignPartitionsCommand.PartitionReassignmentState(seq(1, 2, 3), seq(1, 2, 4), false));
+        states.put(new TopicPartition("bar", 0),
+            new ReassignPartitionsCommand.PartitionReassignmentState(seq(1, 2, 3), seq(1, 2, 4), false));
+
+        assertEquals(
+            "Status of partition reassignment:" + System.lineSeparator() +
+            "Reassignment of partition bar-0 is still in progress." + System.lineSeparator() +
+            "Reassignment of partition foo-0 is completed." + System.lineSeparator() +
+            "Reassignment of partition foo-1 is still in progress.",
+            partitionReassignmentStatesToString(asScala(states)));
+    }
```

Comment:
What about to use `String.join` instead of manual concatenation with line separator. here and below.
```suggestion
        assertEquals(String.join(System.lineSeparator(),
            "Status of partition reassignment:",
            "Reassignment of partition bar-0 is still in progress.",
            "Reassignment of partition foo-0 is completed.",
            "Reassignment of partition foo-1 is still in progress."),
            partitionReassignmentStatesToString(asScala(states)));
```

Reply:
OK. Fixed.